### PR TITLE
[dpc-3630] Jd me/dpc 3630 add dpc portal

### PR DESCRIPTION
**Pull request by MEspositoE14s**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3630

## 🛠 Changes

Setup framework for the new dpc-portal app.
Updated docker-compose.portals to start new app.
Added portal DB and connected it to app.
Updated make targets to start, stop and test new app.
Updated Git workflow to run doc-portal tests.

## ℹ️ Context for reviewers

We're building a new doc-portal that will replace dpc-web and dpc-admin, and this is the first step.
We used the same Gemfile as is used in the other rails services, as they now use rails 7. We left out the ActionCable gem, as it had been removed because of a security issue in rails 5, but we could simplify the Gemfile (and thus future upgrades) by putting it back in. We also left in other gems, like Devise (authentication) and fihr for their potential future use, but maybe they should be removed?

## ✅ Acceptance Validation

The portal runs locally, serves up a "Hello World" page, and all tests pass locally and on GitHub.

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: dpcv1-portals-test.sh [hunk 1]
```diff
@@ -33,6 +34,14 @@ echo "└───────────────────────
 docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "rubocop" dpc_admin
 docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "bundle exec rails spec" dpc_admin
 
+echo "┌───────────────────────────┐"
+echo "│                           │"
+echo "│  Running DPC Portal Tests │"
+echo "│                           │"
+echo "└───────────────────────────┘"
+docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "rubocop" dpc_portal
+docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "bundle exec rails spec" dpc_portal
```
Comment (jdettmannnava) [inline dpcv1-portals-test.sh:43]: I prefer "bundle exec rspec" to "bundle exec rails spec"
See https://github.com/rspec/rspec-rails/issues/208#issuecomment-413114

File: dpcv1-portals-test.sh [hunk 2]
```diff
@@ -33,6 +34,14 @@ echo "└───────────────────────
 docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "rubocop" dpc_admin
 docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "bundle exec rails spec" dpc_admin
 
+echo "┌───────────────────────────┐"
+echo "│                           │"
+echo "│  Running DPC Portal Tests │"
+echo "│                           │"
+echo "└───────────────────────────┘"
+docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "rubocop" dpc_portal
```
Comment (jdettmannnava) [inline dpcv1-portals-test.sh:42]: I think replace "rubocop" with "bundle exec rubocop" just for consistency

File: README.md [hunk 3]
```diff
@@ -59,15 +59,16 @@ providers to deliver high quality care directly to Medicare beneficiaries. See
...
-|Service|Type|Description|Stack|
-|---|---|---|---|
-|[dpc-web](/dpc-web)|Public Portal|Portal for managing organizations|Ruby on Rails|
-|[dpc-admin](/dpc-admin)|Internal Portal|Administrative Portal for managing organizations|Ruby on Rails|
-|[dpc-api](/dpc-api)|Public API|Asynchronous FHIR API for managing organizations and requesting or retrieving data|Java (Dropwizard)|
-|[dpc-attribution](/dpc-attribution)|Internal API|Provides and updates data about attribution|Java (Dropwizard)|
-|[dpc-consent](/dpc-consent)|Internal API|Provides and updates information about data-sharing consent for individuals|Java (Dropwizard)|
-|[dpc-queue](/dpc-queue)|Internal API|Provides and updates data about export jobs and batches|Java (Dropwizard)|
-|[dpc-aggregation](/dpc-aggregation)|Internal Worker Service|Polls for job batches and exports data for singular batches|Java (Dropwizard + RxJava)|
+| Service                             |Type| Description                                                                                |Stack|
```
[the remark this answers is not in the document]
Comment (MEspositoE14s) [reply README.md]: This was IntelliJ trying to format things to make them look better.  I yanked them.

File: README.md [hunk 4]
```diff
@@ -409,7 +410,7 @@ JavaDoc tool. The [Intelli-J Idea](https://jetbrains.com/idea) integrated develo
 ## Building the Additional Services
 ###### [`^`](#table-of-contents)
 
-Documentation on building the DPC Website is covered in the specific [README](dpc-web/README.md).
```
[the remark this answers is not in the document]
Comment (MEspositoE14s) [reply README.md:412]: I stuck it back in.  We can always pull it later when we eventually get rid of dpc-web and dpc-admin.

File: dpc-portal-test.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml up start_core_dependencies
+docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "bundle exec rails db:create db:migrate RAILS_ENV=test" dpc_portal
+
+# Run the tests
+echo "┌───────────────────────────┐"
+echo "│                           │"
+echo "│  Running DPC Portal Tests │"
+echo "│                           │"
+echo "└───────────────────────────┘"
+docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "rubocop" dpc_portal
```
[the remark this answers is not in the document]
Comment (MEspositoE14s) [reply dpc-portal-test.sh]: Fine by me.  I also made the change in dpc-web-portal-test.sh and dpc-admin-portal-test.sh.

File: dpc-portal-test.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+docker-compose -p start-v1-portals -f docker-compose.yml -f docker-compose.portals.yml run --entrypoint "bundle exec rails spec" dpc_portal
```
[the remark this answers is not in the document]
Comment (MEspositoE14s) [reply dpc-portal-test.sh]: Fine by me.  I also made the change in dpc-web-portal-test.sh and dpc-admin-portal-test.sh.

File: dpcv1-portals-test.sh (hunk 1 above)
Comment (MEspositoE14s) [reply dpcv1-portals-test.sh]: Made the change in all of our scripts.

File: dpcv1-portals-test.sh (hunk 2 above)
Comment (MEspositoE14s) [reply dpcv1-portals-test.sh]: Made the change in all of our scripts.

File: dpc-portal/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+# Copy the code, test the app, and build the assets pipeline
+COPY /dpc-portal /dpc-portal
+
+RUN set -eux \
+    & apk add \
+        --no-cache \
+        nodejs \
+        yarn
+#RUN rails webpacker:install
+#RUN RAILS_ENV=production WEBPACKER_NODE_MODULES_BIN_PATH="node_modules/.bin" SECRET_KEY_BASE=dummy bundle exec rake assets:precompile --trace
```
[the remark this answers is not in the document]
Comment (MEspositoE14s) [reply dpc-portal/Dockerfile]: Looks like you got this working!

File: dpc-portal/README.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# README
+
+This is the Data Point of Care (DPC) website. It will allow users to sign up for the DPC service, get credentialed and configure their account.
+
+# Running via Docker
+
+The DPC website can be run locally via docker. Follow the below steps to build and run the website into a Docker container.
+
+- Run `make start-portals` to start dpc-portal in a docker container.
+  - This will also start dpc-web and dpc-admin, which are no longer used in production.
```
Comment (kyeah) [inline dpc-portal/README.md:10]: Would suggest rewording this a bit since these apps were never used in production 👋 

```suggestion
  - This will also start dpc-web and dpc-admin, which are only used in the sandbox environment.
```

**Review by ashley-weaver — APPROVED** [review]
LGTM!

**Review by kyeah — APPROVED** [review]
Worked for me, everything looks good :) Thank you!

> We left out the ActionCable gem, as it had been removed because of a security issue in rails 5, but we could simplify the Gemfile (and thus future upgrades) by putting it back in.

Let's keep this piecemeal for now 😄 maybe we'll want to switch in the future if upgrades are a bit cumbersome.

> We also left in other gems, like Devise (authentication) and fihr for their potential future use, but maybe they should be removed?

We have some upcoming tickets where these gems might be helpful so let's leave them in, we can pull them out later if we end up not using them.
